src/components/TextInput/index.jsx

```diff
@@ -1,7 +1,46 @@
-import React from "react";
+import React, { useState } from "react";
+import styles from "./styles/styles.module.scss";
 
-const TextInput = () => {
-  return <p>text input</p>;
+const TextInput = ({ label, minlength, isRequired = true }) => {
+  // let [inputStyle, setInputStyle] = useState(styles.input);
+  let [errorText, setErrorText] = useState("");
+
+  const handleInputValue = (event) => {
+    if (event.target.value.length === 0) {
+      setErrorText("can't be empty");
+    } else if (event.target.value.length < 2) {
+      setErrorText("must be at least 2 characters long");
+    } else {
+      setErrorText("");
+    }
+  };
+
+  let textInput = isRequired ? (
+    <label className={styles.label}>
+      {label}
+      <input
+        className={`${styles.input} ${errorText ? styles.error : ""}`}
+        type="text"
+        minlength={minlength}
+        required
+        onBlur={handleInputValue}
+      />
+      {errorText && <p className={styles.errorText}>{errorText}</p>}
+    </label>
+  ) : (
+    <label className={styles.label}>
+      {label}
+      <input
+        className={`${styles.input} ${errorText ? styles.error : ""}`}
+        type="text"
+        minlength={minlength}
+        onBlur={handleInputValue}
+      />
+      {errorText && <p className={styles.errorText}>{errorText}</p>}
+    </label>
+  );
```

Comment:
This if and else block has just one thing that is different from what I can see. Please make that one thing only conditional. (`required` attribute in your case).

In general: When you see a repeated code, most probably, you can avoid that.

Reply:
doneee